src/mdel/trainer.py

```diff
@@ -236,9 +255,21 @@ def main():
         model_args, data_args, training_args = parser.parse_args_into_dataclasses()
 
     # Sending telemetry. Tracking the example usage helps us better allocate resources to maintain them. The
-    # information sent is the one passed as arguments along with your Python/PyTorch versions.
+    # information sent is the one passed as arguments along with your
+    # Python/PyTorch versions.
     send_example_telemetry("run_clm", model_args, data_args)
 
+    if training_args.report_to[0] == "wandb" and training_args.local_rank == 0:
+        import wandb
+
+        wandb_name = model_args.model_name_or_path
+        wandb.init(
+            project="mixture",
```

Comment:
we can leave project, entity and name empty, as we define them with env variables in the bootstrap shell script

Reply:
For a pr I am working on that will include the wandb URL in model card and fix a yaml bug in model card I no longer set wandb variables in the shell script so that I can easily access the url

```
@dataclass
class UploadArguments:

    wandb_project: Optional[str] = field(
        default=None,
        metadata={
            "help": (
                "The wandb project name to upload the model card to."
            )
        },
    )

    wandb_run_name: Optional[str] = field(
        default=None,
        metadata={
            "help": (
                "The wandb run name to upload the model card to."
            )
        },
    )
    
    wandb_entity: Optional[str] = field(
        default=None,
        metadata={
            "help": (
                "The wandb entity name to upload the model card to."
            )
        },
    )
```

Reply:
You can make default=os.getenv(...), so that it will be backward compatible

Reply:
I added your suggestion and created a branch. I have not tested wandb upload with the most recent changes so I am refraining from opening a pr.  https://github.com/ontocord/MDEL/tree/repo-card-wandb